eZ/Bundle/EzPublishCoreBundle/EventListener/RejectExplicitFrontControllerRequestsListener.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * This behavior is reflected in meta repository's vhost.template so it should
+ * not be triggered on recommended nginx/apache setups. It mostly applies to
+ * Platform.sh and setups not relying on recommended vhost configuration.
+ */
+class RejectExplicitFrontControllerRequestsListener implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return [
+            KernelEvents::REQUEST => [
+                ['onKernelRequest', 255],
+            ],
+        ];
+    }
+
+    /**
+     * @param \Symfony\Component\HttpKernel\Event\GetResponseEvent $event
+     */
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        if ($event->getRequestType() !== HttpKernelInterface::MASTER_REQUEST) {
+            return;
+        }
+
+        $request = $event->getRequest();
+        $scriptFileName = preg_quote(basename($request->getScriptName()), '\\');
```

Comment:
Could you expand on "better compatibility" please, for educational purposes?
I was going to say that basename() isn't needed anymore, but it seems that SCRIPT_NAME includes the leading slash, so let's keep it.
(Both SCRIPT_NAME and SCRIPT_FILENAME are injection safe, from what I've read.)

Reply:
If you look at the implementation of `Request::getScriptName` you will see that it's resolved from `SCRIPT_NAME` or `ORIG_SCRIPT_NAME` for compatibility with 1and1 hosting provider. I think it's safer to use it this way.